x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -874,54 +885,32 @@ export function overridePackageInputs(
         ),
         package: { name: packageName, version: basePackagePolicy.package!.version },
       };
+
       if (dryRun) {
         errors.push({
           key: override.type,
           message: String(e.error),
         });
         continue;
-      } else throw e;
+      } else {
+        throw e;
+      }
     }
 
     if (typeof override.enabled !== 'undefined') originalInput.enabled = override.enabled;
     if (typeof override.keep_enabled !== 'undefined')
       originalInput.keep_enabled = override.keep_enabled;
 
     if (override.vars) {
-      try {
```

Comment:
A lot of this inline error handling can be removed because we delegate validation error checking to `validatePackagePolicy` now.